src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -821,10 +821,12 @@ static int TopicPartition_clear (TopicPartition *self) {
 
 static void TopicPartition_setup (TopicPartition *self, const char *topic,
 				  int partition, long long offset,
+				  char *metadata,
 				  rd_kafka_resp_err_t err) {
 	self->topic = strdup(topic);
 	self->partition = partition;
 	self->offset = offset;
+	self->metadata = metadata;
```

Comment:
I don't think this is safe since it will reference the argument object without holding a reference. Suggest copying it and making the parameter const.